src/libraries/Hooks.sol

```diff
@@ -233,15 +248,19 @@ library Hooks {
     /// @notice calls beforeSwap hook if permissioned and validates return value
     function beforeSwap(IHooks self, PoolKey memory key, IPoolManager.SwapParams memory params, bytes calldata hookData)
         internal
-        returns (int256 amountToSwap, int128 hookDeltaSpecified)
+        returns (int256 amountToSwap, int128 hookDeltaSpecified, uint24 swapFee)
     {
         amountToSwap = params.amountSpecified;
-        if (msg.sender == address(self)) return (amountToSwap, hookDeltaSpecified);
+        swapFee = type(uint24).max;
+        uint24 _swapFee;
+        if (msg.sender == address(self)) return (amountToSwap, hookDeltaSpecified, swapFee);
         if (self.hasPermission(BEFORE_SWAP_FLAG)) {
-            hookDeltaSpecified = self.callHookWithReturnDelta(
+            (hookDeltaSpecified, _swapFee) = self.callHookWithReturnDeltaAndFee(
                 abi.encodeWithSelector(IHooks.beforeSwap.selector, msg.sender, key, params, hookData),
                 self.hasPermission(BEFORE_SWAP_RETURNS_DELTA_FLAG)
-            ).toInt128();
+            );
+
+            if (key.fee.isDynamicFee()) swapFee = _swapFee;
```

Comment:
could you do something similar to how we do hookDeltaSpecified? where we pass in a bool into `callHookWithReturnDeltaAndFee` for if the hook has permission to set the dynamic fee otherwise return 0